airflow/metrics/datadog_logger.py

```diff
@@ -51,7 +55,33 @@
     )
 
 
-class SafeDogStatsdLogger:
+def prepare_stat_with_tags(fn: T) -> T:
+    """Prepare tags and stat."""
+
+    @wraps(fn)
+    def wrapper(self, stat: str | None = None, *args, tags: dict[str, str] | None = None, **kwargs):
+        stat = stat or ""
+
+        if tags and self.metrics_tags:
+            valid_tags: dict[str, str] = {}
+            for k, v in tags.items():
+                if self.metric_tags_validator.test(k):
+                    if all(c not in [",", "="] for c in f"{v}{k}"):
```

Comment:
Why do we care about `=` being in the key or value? Isn't the separator for datadog tags `:` so this should be
 ```suggestion
                     if ":" not in f"{v}{k}":
```

Reply:
Shouldn't this be the job of `self.metric_tags_validator` though?

Reply:
As far as I know, it's not checking it that's why we have the same checks in Statd.

Reply:
But you're right I change the checks for "," and "=" to ":"